### PR TITLE
[feg] Fixed all linter errors

### LINKED ISSUE #6111: [feg][Golang][Lint] Fix linter issues triggered by make lint at feg
Fix all the lint issues created by enabling go linter at #6110 

land this before https://github.com/magma/magma/pull/6109
```
-> golangci-lint already installed
golangci-lint run
plmn_filter/plmn_filter.go:44:4: S1005: unnecessary assignment to the blank identifier (gosimple)
			val, _ := plmnIds[plmnid5]
			^
services/aaa/store/in_memory.go:263:54: unnecessary conversion (unconvert)
				if atomic.CompareAndSwapPointer((*unsafe.Pointer)(&ms.cleanupTimerCtx), unsafe.Pointer(ctx), nil) {
				                                                 ^
services/aaa/store/in_memory.go:172:3: S1005: unnecessary assignment to the blank identifier (gosimple)
		s, _ = st.sm[sid]
		^
services/aaa/store/in_memory.go:182:3: S1005: unnecessary assignment to the blank identifier (gosimple)
		sid, _ = st.sids[imsi]
		^
services/session_proxy/credit_control/gx/config.go:65: File is not `gofmt`-ed with `-s` (gofmt)
			&diameter.DiameterServerConfig{
services/session_proxy/credit_control/gx/gx_client.go:380:83: unnecessary conversion (unconvert)
		m.NewAVP(avp.TGPPMSTimeZone, avp.Vbit, diameter.Vendor3GPP, datatype.OctetString(datatype.OctetString(string([]byte{timezone, 0}))))
		                                                                                ^
services/session_proxy/credit_control/gx/gx_client.go:523:13: unnecessary conversion (unconvert)
	return byte(encodedTimezone)
	           ^
services/testcore/hss/servicers/test_utils/test_subscribers.go:99: File is not `gofmt`-ed with `-s` (gofmt)
			ApnConfig: []*protos.APNConfiguration{&protos.APNConfiguration{
tools/gw_s6a_service_cli/main.go:83:3: S1011: should replace loop with `req.UserId = append(req.UserId, flag.Args()...)` (gosimple)
		for _, uid := range flag.Args() {
		^
mconfig_test/mconfig_test.go:309: File is not `gofmt`-ed with `-s` (gofmt)
					&fegmcfg.DiamClientConfig{
services/envoy_controller/servicers/envoy_controller_test.go:275:7: ineffectual assignment to `err` (ineffassign)
	ret, err := srv.AddUEHeaderEnrichment(ctx, addUe1Rule1)
	     ^
services/envoy_controller/servicers/envoy_controller_test.go:293:7: ineffectual assignment to `err` (ineffassign)
	ret, err := srv.AddUEHeaderEnrichment(ctx, addUe1Rule1)
	     ^
services/envoy_controller/servicers/envoy_controller_test.go:295:7: ineffectual assignment to `err` (ineffassign)
	ret, err = srv.AddUEHeaderEnrichment(ctx, addUe2Rule2)
	     ^
services/s6a_proxy/servicers/s6a_proxy.go:211:38: `occured` is a misspelling of `occurred` (misspell)
			HealthMessage: fmt.Sprintf("Error occured while retrieving health metrics: %s", err),
			                                  ^
services/hlr_proxy/client_api.go:47:3: SA1006: printf-style function with dynamic format string and no further arguments should use print-style function instead (staticcheck)
		log.Printf(errMsg)
		^
services/testcore/ocs/mock_ocs/ocs.go:239:22: `Remaing` is a misspelling of `Remaining` (misspell)
		glog.V(2).Infof("\tRemaing credit for IMSI: %s", imsi)
		                   ^
services/testcore/ocs/mock_ocs/ccr_handler.go:313:6: S1009: should omit nil check; len() for nil slices is defined as zero (gosimple)
		if restrict_rules == nil || len(restrict_rules) == 0 {
		   ^
services/testcore/ocs/mock_ocs/ccr_handler.go:185:5: SA4003: no value of type uint64 is less than 0 (staticcheck)
	if result < 0 {
	   ^
services/csfb/servicers/decode/message/decoder.go:148:2: ineffectual assignment to `readIdx` (ineffassign)
	readIdx += decode.LengthRejectCause
	^
services/csfb/servicers/csfb.go:157:39: `signalling` is a misspelling of `signaling` (misspell)
// to indicate the existence of a NAS signalling Connection
                                      ^
services/csfb/servicers/csfb.go:159:12: `signalling` is a misspelling of `signaling` (misspell)
// the NAS signalling Connection has been established after the paging procedure
           ^
services/csfb/servicers/sctp_connection.go:158:2: S1008: should use 'return conn.sendConn != nil' instead of 'if conn.sendConn == nil { return false }; return true' (gosimple)
	if conn.sendConn == nil {
	^
services/s8_proxy/test_init/test_service_init.go:18:38: SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (staticcheck)
	mockPgw, err := mock_pgw.NewStarted(nil, serverAddr)
	                                    ^
services/eap/providers/sim/definitions.go:104:2: SA9004: only the first constant in this group has an explicit type (staticcheck)
	Version           byte = 1 // SIM's Supported Version
	^
services/s6a_proxy/servicers/s6a_proxy_test.go:135:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go testLoopF()
	^
services/s6a_proxy/servicers/s6a_proxy_test.go:211:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go func() {
	^
services/s6a_proxy/servicers/s6a_proxy_test.go:282:2: SA2002: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (staticcheck)
	go func() {
	^
services/eap/test/services.go:78:65: `retuns` is a misspelling of `returns` (misspell)
// NoUseSwxProxy - a dummu SwxProxy implementation which always retuns an error & should not be called
                                                                ^
tools/s8_cli/main.go:241:51: `recived` is a misspelling of `received` (misspell)
	// check if message was received but GTP message recived was in fact an error
	                                                 ^
tools/s8_cli/main.go:91:3: S1039: unnecessary use of fmt.Sprint (gosimple)
		fmt.Sprint("Starts s8 proxy checking PGW is alive"))
		^
tools/s8_cli/main.go:140:3: S1039: unnecessary use of fmt.Sprint (gosimple)
		fmt.Sprint("Starts s8 proxy checking PGW is alive"))
		^
tools/s8_cli/main.go:378:38: SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (staticcheck)
	mockPgw, err := mock_pgw.NewStarted(nil, testServerAddr)
	                                    ^
gtp/gtp_client_test.go:105:30: `differnt` is a misspelling of `different` (misspell)
	// create same session with differnt QCI and different C-Sgw TEID
	                            ^
services/session_proxy/session_proxy/main.go:117:36: `connexions` is a misspelling of `connections` (misspell)
	glog.Info("------ Create diameter connexions ------")
	                                  ^
services/session_proxy/session_proxy/main.go:187:47: `connexions` is a misspelling of `connections` (misspell)
	glog.Infof("------ Done creating %d diameter connexions ------", totalLen)
	                                             ^
diameter/diameter_client.go:89:32: `relavent` is a misspelling of `relevant` (misspell)
// Input: clientCfg containing relavent diameter settings
                               ^
services/testcore/pcrf/mock_pcrf/ccr_handler.go:106:3: ineffectual assignment to `avps` (ineffassign)
		avps := []*diam.AVP{}
		^
services/testcore/pcrf/mock_pcrf/mock.go:87:2: S1031: unnecessary nil check around range (gosimple)
	if monitorInstalls != nil {
	^
services/testcore/pcrf/mock_pcrf/pcrf.go:408:2: S1031: unnecessary nil check around range (gosimple)
	if monitorInstalls != nil {
	^
services/testcore/pcrf/mock_pcrf/ccr_handler.go:241:5: SA4003: no value of type uint64 is less than 0 (staticcheck)
	if result < 0 {
	   ^
services/envoy_controller/servicers/envoy_controller.go:19: File is not `goimports`-ed (goimports)
	"magma/feg/gateway/services/envoy_controller/control_plane"
services/testcore/hss/client_api_test.go:77:13: ineffectual assignment to `ok` (ineffassign)
	actualErr, ok := status.FromError(err)
	           ^
services/session_proxy/servicers/multiple_session_controller.go:305:55: `ammount` is a misspelling of `amount` (misspell)
		return nil, fmt.Errorf("Index %d is bigger than the ammount of controllers %d", index, len(controllers))
		                                                    ^
services/session_proxy/servicers/policy.go:31:27: `inital` is a misspelling of `initial` (misspell)
// otherwise it sends the inital request to PCRF
                          ^
services/session_proxy/servicers/policy.go:39:35: `inital` is a misspelling of `initial` (misspell)
// sendInitialGxRequest sends the inital request to PCRF. Returns a response
                                  ^
services/session_proxy/servicers/session_controller.go:363:38: `occured` is a misspelling of `occurred` (misspell)
			HealthMessage: fmt.Sprintf("Error occured while retrieving health metrics: %s", err),
			                                  ^
services/session_proxy/servicers/credits.go:91:4: S1023: redundant break statement (gosimple)
			break
			^
services/session_proxy/servicers/policy.go:70:4: S1023: redundant break statement (gosimple)
			break
			^
services/session_proxy/servicers/policy.go:73:4: S1023: redundant break statement (gosimple)
			break
			^
services/session_proxy/servicers/multiple_session_controller.go:289:5: S1002: should omit comparison to bool constant, can be simplified to `!found` (gosimple)
	if found == false {
	   ^
services/session_proxy/servicers/session_controller.go:113:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGy` (gosimple)
	if srv.cfg.DisableGy == false {
	   ^
services/session_proxy/servicers/session_controller.go:321:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGx` (gosimple)
	if srv.cfg.DisableGx == false {
	   ^
services/session_proxy/servicers/session_controller.go:324:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGy` (gosimple)
	if srv.cfg.DisableGy == false {
	   ^
services/session_proxy/servicers/session_controller.go:338:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGx` (gosimple)
	if srv.cfg.DisableGx == false {
	   ^
services/session_proxy/servicers/session_controller.go:344:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGy` (gosimple)
	if srv.cfg.DisableGy == false {
	   ^
services/session_proxy/servicers/session_controller.go:401:5: S1002: should omit comparison to bool constant, can be simplified to `!srv.cfg.DisableGx` (gosimple)
	if srv.cfg.DisableGx == false {
	   ^
services/session_proxy/servicers/session_controller_test.go:589:24: `comming` is a misspelling of `coming` (misspell)
	// This is the answer comming from the second server. NOTE THIS MAY NEED TO BE CHANGED IF idx1 and idx2 are the same
	                      ^
services/session_proxy/servicers/session_controller_test.go:1134:49: `controlers` is a misspelling of `controls` (misspell)
// getMockMultiplexor loads mockMux with random controlers per each imsi and Imsi without prefix and
                                                ^
services/aaa/servicers/responders.go:213:19: `fulfils` is a misspelling of `fulfills` (misspell)
// CancelLocation fulfils S6a's CLR and disconnect UE from AAA if successful
                  ^
services/aaa/servicers/responders.go:229:10: `fulfils` is a misspelling of `fulfills` (misspell)
// Reset fulfils S6a's RSR and disconnect UE from AAA if successful
         ^
services/aaa/servicers/responders.go:268:2: S1017: should replace this `if` statement with an unconditional `strings.TrimPrefix` (gosimple)
	if strings.HasPrefix(imsi, ImsiPrefix) {
	^
services/aaa/servicers/accounting.go:229:13: S1039: unnecessary use of fmt.Sprintf (gosimple)
		errMsg := fmt.Sprintf("Nil AAA Context")
		          ^
services/testcore/vlr/servicers/vlr_test.go:60:2: ineffectual assignment to `nextPair` (ineffassign)
	nextPair, err = srv.GetNextRequestReplyPair()
	^
services/swx_proxy/servicers/multiple_swx_proxy.go:24: File is not `goimports`-ed (goimports)
	"golang.org/x/net/context"
services/swx_proxy/servicers/auth.go:169:66: `cancelled` is a misspelling of `canceled` (misspell)
			err = status.Errorf(codes.Aborted, "MAA for Session ID: %s is cancelled", sid)
			                                                              ^
services/swx_proxy/servicers/multiple_swx_proxy.go:164:55: `ammount` is a misspelling of `amount` (misspell)
		return nil, fmt.Errorf("Index %d is bigger than the ammount of proxies %d", index, len(proxies))
		                                                    ^
services/swx_proxy/servicers/register.go:81:66: `cancelled` is a misspelling of `canceled` (misspell)
			err = status.Errorf(codes.Aborted, "SAA for Session ID: %s is cancelled", sid)
			                                                              ^
services/swx_proxy/servicers/swx_proxy.go:249:38: `occured` is a misspelling of `occurred` (misspell)
			HealthMessage: fmt.Sprintf("Error occured while retrieving health metrics: %s", err),
			                                  ^
services/s8_proxy/servicers/s8_proxy_test.go:234:2: ineffectual assignment to `dsReq` (ineffassign)
	dsReq := &protos.DeleteSessionRequestPgw{Imsi: "000000000000015"}
	^
services/s8_proxy/servicers/s8_proxy.go:152:14: `comming` is a misspelling of `coming` (misspell)
		// address comming from string has precednece
		           ^
services/s8_proxy/servicers/proto_conversions.go:89:14: S1039: unnecessary use of fmt.Sprintf (gosimple)
						Msg:   fmt.Sprintf("Bearer could not be created"),
						       ^
services/s8_proxy/servicers/s8_proxy_test.go:523:38: SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (staticcheck)
	mockPgw, err := mock_pgw.NewStarted(nil, pgwAddrs)
	                                    ^
services/aaa/session_manager/client_api.go:19: File is not `goimports`-ed (goimports)
	"fmt"
service_health/metrics/metrics.go:34:6: S1009: should omit nil check; len() for nil slices is defined as zero (gosimple)
		if metric == nil || len(metric) == 0 {
		   ^
tools/s6a_cli/main.go:294:10: `untill` is a misspelling of `until` (misspell)
	// wait untill all air request are done
	        ^
services/testcore/hss/servicers/rt.go:60:66: `cancelled` is a misspelling of `canceled` (misspell)
			err = status.Errorf(codes.Aborted, "RTA for Session ID: %s is cancelled", sid)
			                                                              ^
services/swx_proxy/servicers/multiple_swx_proxy_test.go:105:17: `addres` is a misspelling of `address` (misspell)
	// returns the addres of the client at SWx side.
	               ^
services/radiusd/collection/metric_aggregate_test.go:108:2: SA4006: this value of `metricAggregate` is never used (staticcheck)
	metricAggregate, err := collection.CreateMetricAggregate(metricName, metricFamily)
	^
tools/swx_cli/main.go:184:3: SA1006: printf-style function with dynamic format string and no further arguments should use print-style function instead (staticcheck)
		fmt.Printf(err.Error())
		^
tools/swx_cli/main.go:199:4: SA1006: printf-style function with dynamic format string and no further arguments should use print-style function instead (staticcheck)
			fmt.Printf(err.Error())
			^
services/radiusd/collection/fetch.go:52:14: S1028: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (gosimple)
		return "", errors.New(fmt.Sprintf("Failed GET request: %s", err))
		           ^
services/radiusd/collection/fetch.go:59:14: S1028: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (gosimple)
		return "", errors.New(fmt.Sprintf("Failed to read GET response body: %s", err))
		           ^
services/radiusd/collection/metric_aggregate.go:55:15: S1028: should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (gosimple)
		return nil, errors.New(fmt.Sprintf("Failed to build metric collector: %s\n", err))
		            ^
services/csfb/servicers/decode/ie/decoder_test.go:162:10: ineffectual assignment to `ieLength` (ineffassign)
	mmInfo, ieLength, err = ie.DecodeVariableLengthIE(wrongChunk3[:3], decode.IELengthMMInformationMin, decode.IEIMMInformation)
	        ^
Makefile:57: recipe for target 'lint' failed
make: *** [lint] Error 1
root@docker-desktop:/magma/feg/gateway#

```

### PR DESCRIPTION
**Pull request by uri200**
Signed-off-by: Oriol Batalla <obatalla@fb.com>

## Summary

Fix all the errors trigger by the linter when we enable it here https://github.com/magma/magma/issues/6110
## Test

make precommit

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: feg/gateway/services/s6a_proxy/servicers/s6a_proxy_test.go [hunk 1]
```diff
@@ -217,10 +217,10 @@ func TestS6aProxyServiceWitPLMNlist(t *testing.T) {
 		}
 
 		// AIR
-		r, err := c.AuthenticationInformation(context.Background(), req)
-		if err != nil {
-			t.Fatalf("GRPC AIR with PLMN IMSI1 Error: %v", err)
-			complChan <- err
+		r, err2 := c.AuthenticationInformation(context.Background(), req)
```
Comment (themarwhal) [inline feg/gateway/services/s6a_proxy/servicers/s6a_proxy_test.go]: should we do `authenticationErr` instead of `err2`? Seems more descriptive

**Review by themarwhal — APPROVED** [review]

File: feg/gateway/services/s6a_proxy/servicers/s6a_proxy_test.go (hunk 1 above)
Comment (uri200) [reply feg/gateway/services/s6a_proxy/servicers/s6a_proxy_test.go]: sure, err2 is ugly

File: feg/gateway/services/aaa/session_manager/client_api.go [hunk 2]
```diff
@@ -17,11 +17,12 @@ package session_manager
...
 	"errors"
 	"fmt"
-	"github.com/golang/glog"
-	"golang.org/x/net/context"
 
 	"magma/feg/gateway/registry"
 	"magma/lte/cloud/go/protos"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
```
Comment (emakeev) [inline feg/gateway/services/aaa/session_manager/client_api.go]: it should be:

```
  import (
	"errors"
	"fmt"

	"github.com/golang/glog"
	"golang.org/x/net/context"

	"magma/feg/gateway/registry"
	"magma/lte/cloud/go/protos"
  )
```

here, just add an empty line after "fmt"

Comment (uri200) [reply feg/gateway/services/aaa/session_manager/client_api.go]: don't we add first magma dependencies and then external dependencies?

File: feg/gateway/services/session_proxy/session_proxy/main.go [hunk 3]
```diff
@@ -68,6 +68,9 @@ func main() {
 	// Add servicers to the service
 	sessionManagerAndHealthServer, err := servicers.
 		NewCentralSessionControllerDefaultMultiplexWithHealth(controllerParms, policyDBClient)
+	if err != nil {
+		glog.Fatalf("Could not add Health Server to servicer: %s", err)
+	}
```
Comment (emakeev) [inline feg/gateway/services/session_proxy/session_proxy/main.go:73]: Does it have to be 'Fatalf' here?
Will this work?

```
	if err != nil {
		glog.Errorf("Could not add Health Server to servicer: %s", err)
	} else {
	        lteprotos.RegisterCentralSessionControllerServer(srv.GrpcServer, sessionManagerAndHealthServer)
	        protos.RegisterServiceHealthServer(srv.GrpcServer, sessionManagerAndHealthServer)
        }	
```

In a single FeG instance, even an 'unhealthy' FeG should mostly function?

**Review by emakeev — APPROVED** [review]

File: feg/gateway/services/aaa/session_manager/client_api.go (hunk 2 above)
Comment (emakeev) [reply feg/gateway/services/aaa/session_manager/client_api.go]: for FeG module, the order was: std pkgs -> third party packages -> magma packages. it's what I see in most of the sources there

File: feg/gateway/services/session_proxy/session_proxy/main.go (hunk 3 above)
Comment (uri200) [reply feg/gateway/services/session_proxy/session_proxy/main.go:73]: Note that `NewCentralSessionControllerDefaultMultiplexWithHealth` returns the single instance or multiple (depending on the config).

So if there is an error,  sessionManagerAndHealthServer will be nil, so no grpc server will be able to be run. The service will be stuclk. 

At least if we crashloop we will be able to see it. And now with this new feature 
 at https://github.com/magma/magma/pull/6288 we may be able to catch those crashes before even landing

**Pull request merged**